### PR TITLE
Fixed the problem with the binary type attribute in the schema

### PR DESCRIPTION
**Pull request by yan-qi**
1. Enabled the binary type attribute in the schema definition (issue: #9)
2. Changed the unit test by adding a binary type attribute
3. Fixed a bug with the Record implementation to make sure the size of a non-fixed-lenght attribute value is written in the record buffer even if it is zero.

Signed-off-by: Yan Qi yan.qi.asu@gmail.com

### CODE REVIEW

**Comment by yan-qi** [comment]
It mainly solved the issue of #9.

**Comment by yan-qi** [comment]
It is recommended to add a comment as "+1" after completing your review.

File: RecordBuffers/src/main/java/datamine/storage/recordbuffers/ReadOnlyRecord.java [hunk 1]
```diff
@@ -308,14 +309,14 @@ private void constructOffsetArrayFromRecordBuffer() {
 					case STRING:
 						// string requires a SHORT to store its length (max < 32k)
 						short arrayLength = bytebuffer.getShort(offset);
-						if (arrayLength > 0) {
+						if (arrayLength >= 0) {
```
Comment (hzhang22) [inline RecordBuffers/src/main/java/datamine/storage/recordbuffers/ReadOnlyRecord.java:312]: Why adding "=" here? Are we going to add 2 to offset even if there is no short value in bytebuffer?

Comment (yan-qi) [inline RecordBuffers/src/main/java/datamine/storage/recordbuffers/ReadOnlyRecord.java:312]: It has been corrected in the master branch.

**Pull request merged**